br/pkg/lightning/mydump/loader.go

```diff
@@ -422,6 +423,10 @@ func (s *mdLoaderSetup) route() error {
 			remainingSchemas = append(remainingSchemas, info)
 		}
 	}
+	// if there are no tables in the exported data, we still need route s.dbSchemas(remainingSchemas)
+	if err := run(remainingSchemas); err != nil {
+		return errors.Trace(err)
+	}
```

Comment:
Suppose we have db1 and two table route rules:
1. db1 -> db2
2. db2 -> db3

This may cause that db1 is recursively routed to db3 while table data is only routed once.